### PR TITLE
add basic release template

### PR DESCRIPTION
**Pull request by minrk**
General coverage of the common steps for the simpler projects
(i.e. not ipython itself, or notebook).

cc @Carreau @takluyver @fperez @willingc @JamiesHQ

### CODE REVIEW

File: docs/source/developer-docs/releasing.rst [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+Making the release
+==================
+
+Now that your changelog is merged, we can actually build and publish the release.
+We'll assume that ``V`` has been declared as a shell variable containing the release version::
+
+    export V=5.1.2
+
+Start by making sure you have a clean checkout of master, with no extra files:
```
Comment (takluyver) [inline docs/source/developer-docs/releasing.rst]: `::`

File: docs/source/developer-docs/releasing.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+Build the distributions::
+
+    python setup.py sdist --formats=gztar,zip
+    python setup.py bdist_wheel
+
+Tag the commit::
+
+    git tag -am "release $V" $V
+
+And finally, publish everything, to github and PyPI using `twine <https://github.com/pypa/twine>`_:
```
Comment (takluyver) [inline docs/source/developer-docs/releasing.rst]: `::`

(GH highlighting in diffs makes it easy to spot where these are missing)

**Comment by takluyver** [comment]
:+1:

File: docs/source/developer-docs/releasing.rst (hunk 2 above)
Comment (minrk) [inline docs/source/developer-docs/releasing.rst]: Fixed, thanks

**Comment by willingc** [comment]
Looks good @minrk. Thanks for reviewing @takluyver. Merging.

An unrelated question: would we ever want an xz format for any Linux distros?

**Comment by takluyver** [comment]
I don't think it's worth dealing with xz - our packages are mostly small enough that the compression algorithm doesn't matter.

**Comment by willingc** [comment]
@takluyver Sounds good to me. Thanks.

**Comment by minrk** [comment]
@willingc I might go the opposite direction, in fact. I think it's debatable in this day and age that we even need two formats. Windows supports tar.gz by default now, the absence of which was I think the main reason for .zip to be included by default. If any of our source dists were big, xz would make sense (notebook will be biggest, due to all of the js, but still only 10MB).

**Comment by Carreau** [comment]
Thanks @willingc ! Happy to have you back !

**Comment by fperez** [comment]
Thanks everyone, great to see this so it becomes ever easier to rotate the release person hat around the team...

**Pull request merged**